app/grandchallenge/algorithms/views.py

```diff
@@ -92,6 +98,75 @@
 logger = logging.getLogger(__name__)
 
 
+class PhaseConfiguredForAlgorithmSubmissionMixin(UserPassesTestMixin):
```

Comment:
A better location for `PhaseConfiguredForAlgorithmSubmissionMixin` and `UserCanSubmitToPhaseMixin` is in the evaluations app, not the algorithms.